examples/Cassie/run_dircon_squatting.cc

```diff
@@ -106,176 +109,178 @@ void DoMain(double duration, int max_iter, string data_directory,
+
+  int num_knotpoints = 20;
+  double min_T = .01 * 19;
+  double max_T = .3 * 19;
   double mu = 1;
-  left_toe_front_constraint.addFixedNormalFrictionConstraints(mu);
-  left_toe_rear_constraint.addFixedNormalFrictionConstraints(mu);
-  right_toe_front_constraint.addFixedNormalFrictionConstraints(mu);
-  right_toe_rear_constraint.addFixedNormalFrictionConstraints(mu);
-
-  const auto& thigh_left = plant.GetBodyByName("thigh_left");
-  const auto& heel_spring_left = plant.GetBodyByName("heel_spring_left");
-  const auto& thigh_right = plant.GetBodyByName("thigh_right");
-  const auto& heel_spring_right = plant.GetBodyByName("heel_spring_right");
-  double rod_length = 0.5012;  // from cassie_utils
-  Vector3d pt_on_heel_spring = Vector3d(.11877, -.01, 0.0);
-  Vector3d pt_on_thigh_left = Vector3d(0.0, 0.0, 0.045);
-  Vector3d pt_on_thigh_right = Vector3d(0.0, 0.0, -0.045);
-  auto distance_constraint_left = DirconDistanceData<double>(
-      plant, thigh_left, pt_on_thigh_left, heel_spring_left, pt_on_heel_spring,
-      rod_length);
-  auto distance_constraint_right = DirconDistanceData<double>(
-      plant, thigh_right, pt_on_thigh_right, heel_spring_right,
-      pt_on_heel_spring, rod_length);
-
-  // get rid of redundant constraint
-  vector<int> skip_constraint_inds;
-  skip_constraint_inds.push_back(3);
-  skip_constraint_inds.push_back(9);
-
-  // Double stance (all four contact points)
-  vector<DirconKinematicData<double>*> double_stance_all_constraint;
-  double_stance_all_constraint.push_back(&left_toe_front_constraint);
-  double_stance_all_constraint.push_back(&left_toe_rear_constraint);
-  double_stance_all_constraint.push_back(&right_toe_front_constraint);
-  double_stance_all_constraint.push_back(&right_toe_rear_constraint);
-  double_stance_all_constraint.push_back(&distance_constraint_left);
-  double_stance_all_constraint.push_back(&distance_constraint_right);
-  auto double_all_dataset = DirconKinematicDataSet<double>(
-      plant, &double_stance_all_constraint, skip_constraint_inds);
-  auto double_all_options =
-      DirconOptions(double_all_dataset.countConstraints(), plant);
-  // Be careful in setting relative constraint, because we skip constraints
-  ///                 || lf    | lr    | rf    | rr      | fourbar
-  /// Before skipping || 0 1 2 | 3 4 5 | 6 7 8 | 9 10 11 | 12 13
-  /// After skipping  || 0 1 2 |   3 4 | 5 6 7 |   8  9  | 10 11
-  double_all_options.setConstraintRelative(0, true);
-  double_all_options.setConstraintRelative(1, true);
-  double_all_options.setConstraintRelative(3, true);
-  double_all_options.setConstraintRelative(5, true);
-  double_all_options.setConstraintRelative(6, true);
-  double_all_options.setConstraintRelative(8, true);
+  auto double_support =
+      DirconMode<double>(evaluators, num_knotpoints, min_T, max_T);
+  double_support.set_mu(mu);
+  // Set x-y coordinates as relative
+  double_support.MakeConstraintRelative(0, 0);  // left_toe, x
```

Comment:
In case someone wants to use this as a reference for creating a DIRCON problem, it might be useful to use the index returned from `evaluators.add_evaluator(&left_toe_eval)` for the first argument.

Reply:
Good idea.